docs/DeveloperGuide.md

```diff
@@ -914,3 +915,24 @@ testers are expected to do more *exploratory* testing.
 
     1. CS2100 TA will be able to view the number of students, number of classes, number of labs.
     2. CS2100 TA will also be able to see the number of labs left to mark.
+  
+  
+## <a name="Effort"></a> **18. Appendix: Effort**
+  
+### Challenges Faced and Achievements Made
+The following is a non-exhaustive list of challenges we encountered during the development of ProgrammerError:
+  1. **Rewriting the Command Syntax** 
+  - This was challenging as it required a nuanced understanding of how CLI syntax is constructed and following standard conventions. 
+  - Not all of us were familiar with Unix syntax, which made it more difficult to design a solution that would be suitable for users who prefer typing.
+  2. **Designing New UI components**
+  - AB3 uses a simple components with an (arguably) poor user interface (UI) and user experience (UX). On the other hand, our team revamped the look and feel of the application, customizing the dark theme specially (using GitHub's color scheme) to suit TAs. This involved much experimentation with color schemes and building new components, such as popup windows to enhance the user experience. 
+  - Other examples include: color-coding the labs which are marked or unmarked (displayed as green or red respectively), as well as customizing the Help window and Dashboard Window to incorporate our new Dark Theme.
+  3. **Working with Third-Party Libraries**
+  - Compared to AB3 which does not use minimal third-party libraries for its features, our team integrated two additional external libraries to develop our `upload` and `download` features. 
```

Comment:
Does not and minimal contradict each other? I think can keep one of them

Reply:
Fix in latest commit